server.ts

```diff
@@ -58,9 +58,9 @@ app.get("/", (req, res) => {
   } else res.redirect("/survey");
 });
 
-const getsurvey = async (urlstr: string, req: Request<{}>, res: Response<any>) =>  {
+const getsurvey = async (query: string | ParsedQs, req: Request<{}>, res: Response<any>) =>  {
   try {
-    const survey_url = new URL(urlstr);
+    const survey_url = new URL(query['url']);
```

Comment:
Just wondering, is there a reason `query.url` is does not work here? Does that fail differently or something?

Reply:
Typescript doesn't know that "url" is in the query object. I could define a set object with a URL, but I think the lookup `['url']` is just simpler

Reply:
Ah, right. I guess we might eventually have a check for it, but I think this is fine for now.